drivers/gpio/CMakeLists.txt

```diff
@@ -1,6 +1,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 zephyr_library()
+zephyr_library_include_directories(${ZEPHYR_BASE}/drivers)
```

Comment:
I'd make this `zephyr_library_include_directories_ifdef`.  We should limit this to what needs it.